queenbee/io/common.py

```diff
@@ -0,0 +1,133 @@
+"""common objects between different IO files."""
+from enum import Enum
+from typing import Dict, List, Any
+
+from pydantic import constr, Field, validator
+from jsonschema import validate as json_schema_validator
+
+from .reference import FolderReference, FileReference, references_from_string
+from ..base.basemodel import BaseModel
+
+
+class ItemType(str, Enum):
+    """Type enum for items in a list."""
+    String = 'string'
+    Integer = 'integer'
+    Number = 'number'
+    Boolean = 'boolean'
+    Folder = 'folder'
+    Array = 'array'
+    Object = 'object'
+
+
+class GenericInput(BaseModel):
+    """Base class for all input types."""
+
+    type: constr(regex='^GenericInput$') = 'GenericInput'
+
+    name: str = Field(
+        ...,
+        description='Input name.'
+    )
+
+    description: str = Field(
+        None,
+        description='Optional description for input.'
+    )
+
+    default: Any = Field(
```

Comment:
Can this "Any" changed to object?

Reply:
This is a placeholder and will be overwritten by every IO type. You should never get an input with such default but I can change it to String if that is better.

Reply:
@MingboPeng, do you think it will be a problem in the baseclass? I can remove it and copy the validator to all the subclasses. There will be some duplicated code but other than that it should be fine.

Reply:
I am not sure what this Any will be translated to in OpenApi. Let's keep it for now and get back to this if it is an issue